### PR TITLE
Task/unit tests

### PR DESCRIPTION
**Pull request by LeonBrands**
Introduces unit tests for all currently available LLRI features

### CODE REVIEW

File: applications/unit_tests/detail/adapter.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            SUBCASE("[Incorrect usage] supported == nullptr")
+            {
+                CHECK_EQ(adapter->queryExtensionSupport(llri::adapter_extension_type::MaxEnum, nullptr), llri::result::ErrorInvalidUsage);
+            }
+
+            SUBCASE("[Correct usage] invalid extension type")
+            {
+                bool supported = false;
+                CHECK_EQ(adapter->queryExtensionSupport(static_cast<llri::adapter_extension_type>(UINT_MAX), &supported), llri::result::Success);
+            }
```
Comment (GlynLeine) [inline applications/unit_tests/detail/adapter.cpp:73]: shouldn't supported also always be false in this case? idk if any extension type would be
```cpp
std::numeric_limits<unsigned int>::max();
```

File: applications/unit_tests/detail/device.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+namespace llri = legion::graphics::llri;
+
+#include <doctest/doctest.h>
+
+TEST_SUITE("Device")
+{
+    TEST_CASE("Instance::createDevice()")
+    {
+        llri::Instance* instance;
+        const llri::instance_desc desc{ 0, nullptr, "", {} };
```
Comment (GlynLeine) [inline applications/unit_tests/detail/device.cpp:11]: isn't this the same as:
```cpp
const llri::instance_desc desc{};
```

File: applications/unit_tests/detail/device.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            instance->destroyDevice(device);
+        }
+
+        llri::destroyInstance(instance);
+    }
+
+    TEST_CASE("Instance::destroyDevice()")
+    {
+        llri::Instance* instance;
+        const llri::instance_desc desc{ 0, nullptr, "", {} };
```
Comment (GlynLeine) [inline applications/unit_tests/detail/device.cpp:78]: same story as before

File: applications/unit_tests/detail/instance.cpp [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    {
+        SUBCASE("[Correct usage] instance == nullptr")
+        {
+            CHECK_NOTHROW(llri::destroyInstance(nullptr));
+        }
+
+        SUBCASE("[Correct usage] instance != nullptr")
+        {
+            llri::Instance* instance;
+            const llri::instance_desc desc{ 0, nullptr, "", {} };
```
Comment (GlynLeine) [inline applications/unit_tests/detail/instance.cpp:91]: same as before

File: applications/unit_tests/detail/instance.cpp [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            REQUIRE_EQ(llri::createInstance(desc, &instance), llri::result::Success);
+
+            CHECK_NOTHROW(llri::destroyInstance(instance));
+        }
+    }
+
+    TEST_CASE("Instance::enumerateAdapters")
+    {
+        llri::Instance* instance;
+        const llri::instance_desc desc{ 0, nullptr, "", {} };
```
Comment (GlynLeine) [inline applications/unit_tests/detail/instance.cpp:101]: same story again

**Review by GlynLeine — APPROVED** [review]

File: applications/unit_tests/detail/adapter.cpp (hunk 1 above)
Comment (LeonBrands) [reply applications/unit_tests/detail/adapter.cpp:73]: Good point! The check is intended to check if a very invalid extension type wouldn't fail the function, it should return false too, I'll add the check

File: applications/unit_tests/detail/device.cpp (hunk 2 above)
Comment (LeonBrands) [reply applications/unit_tests/detail/device.cpp:11]: True

**Review by GlynLeine — APPROVED** [review]

**Pull request merged**